### DISCUSSION TITLE
Unable to download model silero_te

### DISCUSSION THREAD

**Discussion by MariyaTikhonova** [snakers4/silero-models#125] [General]
I'm trying to use your model for repunctuazation. However, I'm unable to run the getting started example as soon as it fells down with the error RuntimeError: [enforce fail at inline_container.cc:222] . file not found: v2_4lang_q/version. How can I fix it? Or where can I download the required model?

model, example_texts, languages, punct, apply_te = torch.hub.load(repo_or_dir='snakers4/silero-models',
                                                                  model='silero_te')

Downloading: "https://github.com/snakers4/silero-models/archive/master.zip" to /home/jovyan/.cache/torch/hub/master.zip
100%
87.5M/87.5M [00:01<00:00, 74.8MB/s]
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/tmp/ipykernel_3514/633877077.py in <module>
      1 model, example_texts, languages, punct, apply_te = torch.hub.load(repo_or_dir='snakers4/silero-models',
----> 2                                                                   model='silero_te')

~/.imgenv-mashkka-gpu-0/lib/python3.7/site-packages/torch/hub.py in load(repo_or_dir, model, *args, **kwargs)
    368         verbose (bool, optional): If ``False``, mute messages about hitting
    369             local caches. Note that the message about first download cannot be
--> 370             muted. Does not have any effect if ``source = 'local'``.
    371             Default is ``True``.
    372         skip_validation (bool, optional): if ``False``, torchhub will check that the branch or commit

~/.imgenv-mashkka-gpu-0/lib/python3.7/site-packages/torch/hub.py in _load_local(hubconf_dir, model, *args, **kwargs)
    397         repo_or_dir = _get_cache_or_reload(repo_or_dir, force_reload, verbose, skip_validation)
    398 
--> 399     model = _load_local(repo_or_dir, model, *args, **kwargs)
    400     return model
    401 

~/.cache/torch/hub/snakers4_silero-models_master/hubconf.py in silero_te()
    134                                        progress=True)
    135 
--> 136     imp = package.PackageImporter(model_path)
    137     model = imp.load_pickle("te_model", "model")
    138     example_texts = model.examples

~/.imgenv-mashkka-gpu-0/lib/python3.7/site-packages/torch/package/package_importer.py in __init__(self, file_or_buffer, module_allowed)
     73             self.filename = str(file_or_buffer)
     74             if not os.path.isdir(self.filename):
---> 75                 self.zip_reader = torch._C.PyTorchFileReader(self.filename)
     76             else:
     77                 self.zip_reader = DirectoryReader(self.filename)

RuntimeError: [enforce fail at inline_container.cc:222] . file not found: v2_4lang_q/version

**Comment by snakers4** [comment]
Hi,

I checked that it works in colab, both for the long and the short example.

> file not found: v2_4lang_q/version

I do not know how it got borked, but the actual file name should be `v2_4lang_q.pt` instead of `v2_4lang_q/version`.
![image](https://user-images.githubusercontent.com/12515440/153234089-03e2f3b1-c719-4077-9b4a-a594de769e35.png)


Does this method work?

```
import os
import yaml
import torch
from torch import package

torch.hub.download_url_to_file('https://raw.githubusercontent.com/snakers4/silero-models/master/models.yml',
                               'latest_silero_models.yml',
                               progress=False)

with open('latest_silero_models.yml', 'r') as yaml_file:
    models = yaml.load(yaml_file, Loader=yaml.SafeLoader)
model_conf = models.get('te_models').get('latest')

# see avaiable languages
available_languages = list(model_conf.get('languages'))
print(f'Available languages {available_languages}')

# and punctuation marks
available_punct = list(model_conf.get('punct'))
print(f'Available punctuation marks {available_punct}')

model_url = model_conf.get('package')

model_dir = "downloaded_model"
os.makedirs(model_dir, exist_ok=True)
model_path = os.path.join(model_dir, os.path.basename(model_url))

if not os.path.isfile(model_path):
    torch.hub.download_url_to_file(model_url,
                                   model_path,
                                   progress=True)

imp = package.PackageImporter(model_path)
model = imp.load_pickle("te_model", "model")
example_texts = model.examples

def apply_te(text, lan='en'):
    return model.enhance_text(text, lan)
```

**Comment by evrrn** [comment]
@MariyaTikhonova

Hi Maria,

could you please verify which Pytorch version you use? I would recommend to check it with torch 1.10.

**Reply by snakers4** [reply]
This error may be also caused by using a PyTorch version before packages were introduced, i.e. < 1.9
